dll/win32/kernel32/client/sysinfo.c

```diff
@@ -393,17 +393,8 @@ GetNumaProcessorNode(IN UCHAR Processor,
  * @param AvailableBytes
  *        The amount of available memory for the node, in bytes.
  *
- * @return If the function succeeds, the return value is non zero.
- *         If the function fails, the return value is zero. To get
- *         extended error information, call GetLastError.
+ * @return TRUE or FALSE on error.You can get extended information using the GetLastError function.
```

Comment:
Wait, what does this function actually return? TRUE/FALSE, or non-zero/zero?
Assuming the previous text was correct, I recommend something more like this.
```
 * @return Returns non-zero value when the function succeeds, or zero when it fails.
           When an error occurs, more information can be retrieved by calling GetLastError.
```

Reply:
`BOOL` is defined as an integer, *int*, with FALSE being 0 whereas TRUE being 1. The TRUE/FALSE wording can be kept however the sentence has to be improved explaining what TRUE and FALSE actually represent (*TRUE or FALSE on error* is incorrect because the function returns FALSE if it fails, it cannot return an either value).

Reply:
@Fraizeraust According to MSDN, it's not returning `TRUE` or `FALSE`, or `1` or `0`, it's returning 'a non-zero number' (that represents how much memory is in free and zeroed pages in the node specified), or `0`.  I doubt we want our documentation to incorrectly indicate that upon success it returns `1` or `TRUE`, when it's far more likely to be some other non-zero number most of the time.

Reply:
Clearly you haven't looked at the function's prototype, did you?
```C
BOOL
WINAPI
GetNumaAvailableMemoryNode(IN UCHAR Node,
                           OUT PULONGLONG AvailableBytes)
```
And this is how `BOOL` is defined in WinDef.h:
```C
#ifndef FALSE
#define FALSE               0
#endif

#ifndef TRUE
#define TRUE                1
#endif

typedef int                 BOOL;
```
The function's code is also pretty self-explanatory about what it does and returns, really.

Reply:
Huh, you're right, the prototype does say `BOOL`, so is MSDN wrong where it says "The GetNumaAvailableMemoryNode function returns the amount of memory consumed by free and zeroed pages on the specified node." or is it supposed to 'return' that value by some other means?

Reply:
Nevermind. Actually looking at the code in the ReactOS implementation, it's C code, and one parameter is IN, and the other is OUT, and that's how it's 'returning' the amount while actually `return`ing `TRUE`.

derp